docs/source/tools/trigger-service/auth0_example.rst

```diff
@@ -0,0 +1,286 @@
+.. _auth0-user-details: https://auth0.com/docs/users/view-user-details
+
+Start Daml Connect
+~~~~~~~~~~~~~~~~~~
+
+Next, configure the relevant Daml Connect components to use Auth0 as the IAM.
+
+Sandbox
+*******
+
+Start the sandbox using the following command.
+Replace ``JSON_Web_Key_Set`` by the corrseponding URL found in the application settings
+and make sure that the ledger ID matches the one in the Auth0 rule.
+
+.. code-block:: shell
+
+  daml sandbox \
+    --address localhost \
+    --port 6865 \
+    --ledgerid daml-auth0-example-ledger \
+    --wall-clock-time \
+    --auth-jwt-rs256-jwks "JSON_Web_Key_Set"
+
+OAuth 2.0 Middleware
+********************
+
+Start the auth middleware using the following command.
+Replace the client identifier and URL placeholders by the corresponding values found in the application settings
+and make sure that the callback URL matches the allowed callback URL in the application settings.
+The ``--callback`` flag defines the middleware's callback URL as exposed through the reverse proxy.
+
+.. code-block:: shell
+
+  DAML_CLIENT_ID="Client_ID" \
+  DAML_CLIENT_SECRET="Client_Secret" \
+  daml oauth2-middleware \
+    --port 3000 \
+    --oauth-auth "OAuth_Authorization_URL" \
+    --oauth-token "OAuth_Token_URL" \
+    --auth-jwt-rs256-jwks "JSON_Web_Key_Set" \
+    --callback http://localhost:5000/auth/cb
+
+Trigger Service
+***************
+
+Start the trigger service using the following command.
+The ``--auth`` flag defines the middleware's URL prefix as exposed through the reverse proxy,
+similarly the ``--auth-callback`` flag defines the trigger service's callback URL as exposed through the reverse proxy.
+
+.. code-block:: shell
+
+  daml trigger-service \
+    --address localhost \
+    --http-port 4000 \
+    --ledger-host localhost \
+    --ledger-port 6865 \
+    --auth http://localhost:5000/auth \
+    --auth-callback http://localhost:5000/trigger/cb
+
+Configure Web Server
+~~~~~~~~~~~~~~~~~~~~
+
+This guide uses `Nginx <nginx_>`_ as a reverse proxy and web server.
+
+- Configure nginx using the following snippet:
+
+  .. code-block:: nginx
+
+    http {
+      server {
+        listen 5000;
+        server_name localhost;
+        root html;
+
+        location /auth/ {
+          proxy_pass http://localhost:3000/;
+        }
+
+        location /trigger/ {
+          proxy_pass http://localhost:4000/;
+        }
+      }
+    }
+
+  This exposes the auth middleware under the URL ``http://localhost:3000/``
+  and the trigger service under the URL ``http://localhost:4000/``.
+
+- Add the following ``index.html`` to your web root:
+
+  .. code-block:: html
+
+    <!DOCTYPE html>
+    <html>
+      <body>
+        <button onclick="listTriggers()">list triggers</button>
+      </body>
+      <script>
+        async function listTriggers() {
+          const resp = await fetch("http://localhost:5000/trigger/v1/triggers?party=Alice");
```

Comment:
Once we have the disclaimer, I’d refer back to it here and say that because the rule accepts all claims, we can get claims for Alice.